ranger-http/src/main/java/com/flipkart/ranger/http/servicefinderhub/HttpShardedServiceFinderFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 Flipkart Internet Pvt. Ltd.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.ranger.http.servicefinderhub;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.flipkart.ranger.core.finder.ServiceFinder;
+import com.flipkart.ranger.core.finder.serviceregistry.MapBasedServiceRegistry;
+import com.flipkart.ranger.core.finderhub.ServiceFinderFactory;
+import com.flipkart.ranger.core.model.Criteria;
+import com.flipkart.ranger.core.model.Service;
+import com.flipkart.ranger.core.model.ServiceNodeSelector;
+import com.flipkart.ranger.core.model.ShardSelector;
+import com.flipkart.ranger.http.config.HttpClientConfig;
+import com.flipkart.ranger.http.serde.HTTPResponseDataDeserializer;
+import com.flipkart.ranger.http.servicefinder.HttpShardedServiceFinderBuilder;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+
+@Getter
```

Comment:
Why? this is a factory, the only accessible thing should be the build etc methods

Reply:
Addressed. Error when copying code from one file to another.